cmake/OpenSSL.cmake

```diff
@@ -1,15 +1,34 @@
+# Check if a file is a symbolic link (macOS)
+function(is_symlink filepath result)
+    execute_process(
+        COMMAND "test" "-h" "${filepath}"
+        RESULT_VARIABLE return_code
+    )
+    if(return_code EQUAL 0)
+        set(${result} TRUE PARENT_SCOPE)
+    else()
+        set(${result} FALSE PARENT_SCOPE)
+    endif()
+endfunction()
+
 if(NOT DEFINED OPENSSL_CONFIGURED)
     if(APPLE)
-        execute_process(COMMAND brew --prefix openssl@3.1
-                OUTPUT_VARIABLE OPENSSL_PREIX RESULT_VARIABLE OPENSSL_RC
+        execute_process(COMMAND brew --prefix openssl@3
+                OUTPUT_VARIABLE OPENSSL_PREFIX RESULT_VARIABLE OPENSSL_RC
```

Comment:
This has a potential pitfall of being too low (3.0); figuring this out may involve calling and parsing the output of `brew info --json openssl@3` to see what actual version did it resolve to.

Reply:
Fixed through the latest commit.